universal-application-tool-0.0.1/app/assets/javascripts/validation.ts

```diff
@@ -0,0 +1,274 @@
+/** The validation controller provides basic client-side validation of form fields. */
+class ValidationController {
+  static readonly ADDRESS_QUESTION_CLASS = '.cf-question-address';
+  static readonly ENUMERATOR_QUESTION_CLASS = '.cf-question-enumerator';
+  static readonly NAME_QUESTION_CLASS = '.cf-question-name';
+
+  static readonly ENUMERATOR_DELETE_TEMPLATE = 'enumerator-delete-template';
+  static readonly BLOCK_SUBMIT_BUTTON_ID = 'cf-block-submit';
+
+  isAddressValid = true;
+  isEnumeratorValid = true;
+  isNameValid = true;
+
+  /** 
+   * Validating on input will call the validators whenever a field is updated.
+   * The default behavior is to only attempt to validate on submit.
+   * 
+   * Validate on input will also disable the submit button when errors are detected.
+   */
+  validateOnInput = false;

```

Comment:
(question) is this a config flag? I don't see a way to change `validateOnInput`. If so, this should be more `static readonly` with capital letters to match the other constants we use.

Reply:
DONE